rpc: TestClockOffsetInPingRequest failed

rpc.TestClockOffsetInPingRequest [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18030936?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18030936?buildTab=artifacts#/) on release-24.3 @ [563905c072f3d74a969c959b8d29eb7f1408b04a](https://github.com/cockroachdb/cockroach/commits/563905c072f3d74a969c959b8d29eb7f1408b04a):


```
=== RUN   TestClockOffsetInPingRequest
    context_test.go:223: server listen
    context_test.go:230: client dial
E241204 14:53:17.048972 407 2@rpc/peer.go:663  [rnode=1,raddr=127.0.0.1:37365,class=system,rpc] 10  failed connection attempt (last connected 0s ago): operation "conn heartbeat" timed out after 101ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I241204 14:53:17.057869 407 2@rpc/peer.go:589  [rnode=1,raddr=127.0.0.1:37365,class=system,rpc] 11  connection is now healthy (after 0s)
    context_test.go:244: first two pings check
    context_test.go:252: ping 1 check
    context_test.go:260: 
        	Error Trace:	pkg/rpc/context_test.go:260
        	            				pkg/rpc/context_test.go:175
        	Error:      	Should not be zero, but was 0
        	Test:       	TestClockOffsetInPingRequest
        	Messages:   	1
--- FAIL: TestClockOffsetInPingRequest (0.22s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClockOffsetInPingRequest.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/rpc/context_test.go
+++ b/pkg/rpc/context_test.go
@@ -202,7 +202,7 @@ func testClockOffsetInPingRequestInternal(t *testing.T, clientOnly bool) {
 	clientOpts := opts
 	// Experimentally, values below 50ms seem to incur flakiness.
 	clientOpts.RPCHeartbeatInterval = 100 * time.Millisecond
-	clientOpts.RPCHeartbeatTimeout = 100 * time.Millisecond
+	clientOpts.RPCHeartbeatTimeout = 200 * time.Millisecond
 	clientOpts.ClientOnly = clientOnly
 	clientOpts.OnOutgoingPing = func(ctx context.Context, req *PingRequest) error {
 		select {

